dash/development/_jl_components_generation.py

```diff
@@ -23,59 +26,48 @@
 export {funcname}
 
 """
-    {funcname}(;kwargs...)
-    {funcname}(children::Any;kwargs...)
-    {funcname}(children_maker::Function;kwargs...)
+    {funcname}(;kwargs...){childrens_signatures}
 
 {docstring}
 """
 function {funcname}(; kwargs...)
-        available_props = Set(Symbol[{component_props}])
-        wild_props = Set(Symbol[{wildcard_symbols}])
-        wild_regs = r"^(?<prop>{wildcard_names})"
-
-        result = Component("{element_name}", "{module_name}", Dict{{Symbol, Any}}(), available_props, Set(Symbol[{wildcard_symbols}]))
-
-        for (prop, value) = pairs(kwargs)
-            m = match(wild_regs, string(prop))
-            if (length(wild_props) == 0 || isnothing(m)) && !(prop in available_props)
-                throw(ArgumentError("Invalid property $(string(prop)) for component " * "{funcname}"))
-            end
-
-            push!(result.props, prop => Front.to_dash(value))
-        end
-
-    return result
+        available_props = Symbol[{component_props}]
+        wild_props = Symbol[{wildcard_symbols}]
+        return Component("{funcname}", "{element_name}", "{module_name}", available_props, wild_props; kwargs...)
 end
+{childrens_definitions}
+'''  # noqa:E501
 
-function {funcname}(children::Any; kwargs...)
-    result = {funcname}(;kwargs...)
-    push!(result.props, :children => Front.to_dash(children))
-    return result
-end
+jl_childrens_signatures = '''
+    {funcname}(children::Any;kwargs...)
+    {funcname}(children_maker::Function;kwargs...)
+'''
 
+jl_childrens_definitions = '''
+{funcname}(children::Any; kwargs...) = {funcname}(;kwargs..., children = children)
 {funcname}(children_maker::Function; kwargs...) = {funcname}(children_maker(); kwargs...)
-'''  # noqa:E501
+'''
 
 jl_package_file_string = '''
 module {package_name}
-using Dash
+using {base_package}
 
 const resources_path = realpath(joinpath( @__DIR__, "..", "deps"))
 const version = "{version}"
 
 {component_includes}
 
 function __init__()
-    Dash.register_package(
-        Dash.ResourcePkg(
+    DashBase.register_package(
+        DashBase.ResourcePkg(
```

Comment:
Not new here, but is it important that this be two nested calls? Could it be refactored to just:
```julia
DashBase.register_package(
    "{project_shortname}",
    resources_path,
    version = version,
    [
        {resources_dist}
    ]
)
```

Reply:
@waralex points out we can do this later - not a major issue anyway.